modules/ROOT/pages/dataweave-indexed-readers.adoc

```diff
@@ -0,0 +1,51 @@
+= Indexed Readers in DataWeave
+
+DataWeave can read input data (such as XML, JSON, and CSV) in an indexed strategy to avoid loading the whole document in memory while still allowing random access just like in the in-memory strategy. Use indexed readers when the input is larger than a certain configurable threshold or if the `indexedReader` setting is set to `true`.
+
+To manage files that don’t fit in memory, write the files to a temporary file in disk and index the document content beforehand, which also takes time and disk space.
+
+Indexed readers can process input files up to 20GBs. For bigger files use the streaming mode, which is faster. Despite this mode being more limited in functionality, there’s no maximum input size limitation for streaming readers.
+
+The actual maximum input file size supported is difficult to estimate because it depends on the content of the input.
+
+The following list shows the limits on different parts of the input document:
+
+* Max nesting depth: `4096`.
+* Max individual value size: `4GB`.
+* Tokens: `​​2^32-1` (approximately 4 billion). +
+A token is either a key/value, object-start or array-start, which limits the max document size.
+The amount of the max document size is irrespective of the size of each token in the file. The bigger your average value length, the bigger your file size can be because what is important is the amount of tokens, not their size.
```

Comment:
What is the difference between input, document, and input document? Try to standardize the term used to refer to the incoming data.
Ditto maximum input size, maximum input file size, limits on different parts [max] of the input document...and then max nesting, max value size, token limiting max document size.
There seems to be a lot of overlapping similar terminology used in the topic.

Reply:
```suggestion
The max document size is based on the number of tokens, not their size. 
```
"The amount of the max document size" > "The max document size"
"The bigger your average value length" > length of what? "The longer (the length)," not "the bigger the length"